app/controllers/users/two_factor_authentication_controller.rb

```diff
@@ -3,6 +3,8 @@ class TwoFactorAuthenticationController < ApplicationController
     include TwoFactorAuthenticatable
     skip_before_action :handle_two_factor_authentication
 
+    before_action :lock_out_max_2fa_sends
```

Comment:
Call this method inside `handle_valid_otp_delivery_preference` since it only applies to OTP, and not TOTP? A global before action means this logic will get exercised when it doesn't need to. It only needs to be exercised when `handle_valid_otp_delivery_preference` is called.